### PR TITLE
update hashring tests as an example of how to convert to pytest tests

### PR DESCRIPTION
**Pull request by terencehonles**
This PR has been annotated with a brief description how to convert unittest tests to pytest test cases for #521.

### CODE REVIEW

File: tests/test_hashring.py [hunk 1]
```diff
@@ -14,24 +14,21 @@ def __repr__(self):
...
-    def setUp(self):
-        self.node0 = Node(0)
-        self.node1 = Node(1)
-        self.node2 = Node(2)
+@pytest.fixture
+def hash_ring():
+    return HashRing([Node(i) for i in range(3)])
 
-        self.nodes = [self.node0, self.node1, self.node2]
-        self.ring = HashRing(self.nodes)
```
Comment (terencehonles) [inline tests/test_hashring.py:24]: Unfortunately the diff here is not great, but `setUp()` code can generally be converted to a `pytest.fixture`. They can either return a value or yield a value in order to perform cleanup after a test finishes (to act as a `tearDown()`).

For more information: https://docs.pytest.org/en/stable/fixture.html#quick-example

File: tests/test_hashring.py [hunk 2]
```diff
@@ -14,24 +14,21 @@ def __repr__(self):
...
 
-    def test_hashring(self):
-        ids = []
+def test_hashring(hash_ring):
```
Comment (terencehonles) [inline tests/test_hashring.py:22]: The `hash_ring` fixture is requested here and references are adjusted in the test function.

File: tests/test_hashring.py [hunk 3]
```diff
@@ -14,24 +14,21 @@ def __repr__(self):
...
-        self.assertEqual(ids, [0, 2, 1, 2, 2, 2, 2, 0, 1, 1])
+    assert ids == [0, 2, 1, 2, 2, 2, 2, 0, 1, 1]
 
-    def test_hashring_brute_force(self):
-        for key in (f"test{x}" for x in range(10000)):
-            self.ring.get_node(key)
+
+def test_hashring_brute_force(hash_ring):
+    for key in (f"test{x}" for x in range(10000)):
+        assert hash_ring.get_node(key)
```
Comment (terencehonles) [inline tests/test_hashring.py:34]: I'm not sure what the previous test was actually testing other than no exceptions were thrown, but this test now asserts that the value is also truthy.

**Review by WisdomPill — COMMENTED** [review]
Maybe node could be a `dataclass`? They are available from python 3.6 so it should be okay to use them

File: tests/test_hashring.py (hunk 3 above)
Comment (WisdomPill) [reply tests/test_hashring.py:34]: That's quite strange, it was not asserting anything, it can be removed as far as I can tell

**Comment by terencehonles** [comment]
> Maybe `Node` could be a `dataclass`? They are available from python 3.6 so it should be okay to use them

Should we leave that conversion out of the PR since that's starting to distract from the point of this PR (an example of converting unittest tests to pytest tests)?

**Comment by WisdomPill** [comment]
Agree, I am merging it then

**Pull request merged**